cli/pcluster/config/param_types.py

```diff
@@ -495,14 +495,21 @@ def from_cfn_params(self, cfn_params):
 
         return self
 
+    def get_cfn_value(self):
+        """
+        Convert parameter value into CFN value.
+
+        Insignificant trailing zeros removed to correctly match UseSpotPrice with default value
```

Comment:
I'd change the comment with:
```
Insignificant trailing zeros removed to simplify comparison.
```

Reply:
Updated with: "Insignificant trailing zeros removed to correctly match CloudFormation conditions using "0" as test value"